tests/integration/test_mdg_generation.py

```diff
@@ -0,0 +1,282 @@
+
+        with pytest.raises(TypeError) as error_message:
+            grid_type = complex(1, 2)
+            ref_msg = str("grid_type must be str, not %r" % type(grid_type))
+            pp.create_mdg(grid_type, mesh_arguments, fracture_network)
+        assert ref_msg in str(error_message.value)
+
+        with pytest.raises(ValueError) as error_message:
+            grid_type = "Simplex"
+            ref_msg = str(
+                "grid_type must be in ['simplex', 'cartesian', 'tensor_grid'] not %r"
+                % grid_type
+            )
+            pp.create_mdg(grid_type, mesh_arguments, fracture_network)
+        assert ref_msg in str(error_message.value)
+
+    def test_mesh_arguments_inconsistencies(self):
+
+        grid_type = "simplex"
+        fracture_network = self.generate_network(0, [0, 1, 2])
+
+        with pytest.raises(TypeError) as error_message:
+            mesh_arguments = [self.h_reference()]
+            ref_msg = str(
+                "mesh_arguments must be dict[str], not %r" % type(mesh_arguments)
+            )
+            pp.create_mdg(grid_type, mesh_arguments, fracture_network)
+        assert ref_msg in str(error_message.value)
+
+        with pytest.raises(TypeError) as error_message:
+            mesh_size = complex(1, 2)
+            mesh_arguments: dict[str] = {"mesh_size": mesh_size}
+            ref_msg = str("mesh_size must be float, not %r" % type(mesh_size))
+            pp.create_mdg(grid_type, mesh_arguments, fracture_network)
+        assert ref_msg in str(error_message.value)
+
+        with pytest.raises(ValueError) as error_message:
+            mesh_size = -1.0
+            mesh_arguments: dict[str] = {"mesh_size": mesh_size}
+            ref_msg = str("mesh_size must be strictly positive %r" % mesh_size)
+            pp.create_mdg(grid_type, mesh_arguments, fracture_network)
+        assert ref_msg in str(error_message.value)
+
+    def test_network_inconsistencies(self):
+
+        grid_type = "simplex"
+        fracture_network = self.generate_network(0, [0, 1, 2])
+        mesh_arguments: dict[str] = {"mesh_size": self.h_reference()}
+
+        with pytest.raises(ValueError) as error_message:
+            fracture_network.domain.dim = 0
+            ref_msg = str(
+                "Inferred dimension must be 2 or 3, not %r"
+                % fracture_network.domain.dim
+            )
+            pp.create_mdg(grid_type, mesh_arguments, fracture_network)
+            fracture_network.domain.dim = 2
+        assert ref_msg in str(error_message.value)
+
+        with pytest.raises(TypeError) as error_message:
+            ref_msg = str(
+                "fracture_network must be FractureNetwork2d or FractureNetwork3d, not %r"
+                % type(complex(1, 2))
+            )
+            pp.create_mdg(grid_type, mesh_arguments, complex(1, 2))
+        assert ref_msg in str(error_message.value)
+
+
+if __name__ == "__main__":
+    unittest.main()
```

Comment:
I don't think we need this after we moved to the pytest paradigm.

Reply:
Agree, it was removed.